azavea/tilegarden
# Feature/fix tile shearing

## Overview

This PR fixes the warping that was taking place when rendering tiles, which ended up being the result of data being mis-projected:
 * The target srs was not specified properly in `map-config.mml`. Tiles for the demo had to end up in web mercator but they were being projected as WGS84, which was causing the warping. If the source and target srs strings both match the data in the database and the desired output projection respectively, mapnik will automatically re-project the data.
 * Bounding boxes for individual tiles were being calculated in WGS84, which meant that even when tiles were being properly projected as web mercator they were being cropped at the wrong latitude/longitude. I removed the bounding box calculation dependency and wrote a new version that can calculate bounding boxes for any target projection.

### Demo
![image](https://user-images.githubusercontent.com/16990679/42836902-60914dd2-89ca-11e8-8c0e-09e70a1e4adf.png)


## Testing Instructions

 * Run `./scripts/clean`, `./scripts/update`, and `./scripts/server` to start a new development server with the updated data from this PR. I included the `street_centerlines` data set in it's original projection in order to demonstrate mapnik's reprojection capabilities but it won't get added to the database if a similarly named table already exists.
* Open `src/demo/datatype-demo.html` to see the server serve properly projected content.


Resolves #43
